### ISSUE TITLE
Incompatibility with PHPUnit 10 (Pest 2) - overriding final method

### ISSUE THREAD

**Issue by CadenP** [itsgoingd/clockwork#627]
Attempting to run a test with PHPUnit 10 (or Pest 2) which has the `UsesClockwork` trait causes the test to fail with the following message:

```
PHP Fatal error:  Cannot override final method PHPUnit\Framework\Assert::assertThat() in /opt/project/vendor/itsgoingd/clockwork/Clockwork/Support/Laravel/Tests/UsesClockwork.php on line 71
```

According to [PHPUnit's changelog](https://github.com/sebastianbergmann/phpunit/blob/main/ChangeLog-10.0.md#1000---2023-02-03):
> The public methods of `PHPUnit\Framework\Assert` and `PHPUnit\Framework\TestCase` are now `final`

Pest 2 requires PHPUnit 10, so it shares the incompatibility.

**Comment by kdevan** [comment]
+1 here, running into this too. And I love using clockwork with tests!

**Comment by itsgoingd** [comment]
Hey, I wrote some very suspect php code and implemented a new Clockwork extension for PHPUnit 10. I would appreciate if you could try it out, since I don't have any good PHPUnit 10 test-suite for testing.

To try the new extension, update Clockwork to `dev-master`, remove the `UsesClockwork` trait and add following to your `phpunit.xml`:

```xml
<extensions>
    <bootstrap class="Clockwork\Support\Laravel\Tests\ClockworkExtension" />
</extensions>
```

**Comment by CadenP** [comment]
I tested it as well. Works great! Thanks for making this work.

**Comment by walterdis** [comment]
Hi.
Do i have to configure something else after enabling 'collect' => env('CLOCKWORK_TESTS_COLLECT', true), and using the extension?

I'm runing php artisan test and php artisan test --filter="SomeControllerTest" and its not working.

Thanks ;)

**Comment by itsgoingd** [comment]
@walterdis Yeah, adding the PHPUnit extension and setting `CLOCKWORK_TESTS_COLLECT=true` should work. If it doesn't work, can you try updating to the latest master?

**Comment by walterdis** [comment]
> @walterdis Yeah, adding the PHPUnit extension and setting `CLOCKWORK_TESTS_COLLECT=true` should work. If it doesn't work, can you try updating to the latest master?

Worked!!
Thanks a lot ;)

**Comment by itsgoingd** [comment]
Now available in Clockwork 5.2.
